crypto/http/http_client.c

```diff
@@ -304,7 +307,8 @@ int OSSL_HTTP_REQ_CTX_set1_req(OSSL_HTTP_REQ_CTX *rctx, const char *content_type
 
     res = (mem = ASN1_item_i2d_mem_bio(it, req)) != NULL
         && set_content(rctx, content_type, mem);
-    BIO_free(mem);
+    if (!res)
```

Comment:
This is the right fix.
Being at this, please rename `set_content()` to `set0_content()` for pointing out that ownership on `req` is transferred.

Reply:
Another issue:
We now need to call `BIO_free(rctx->req)` in `OSSL_HTTP_REQ_CTX_free()`, 
otherwise there is a mem leak.

Reply:
I do not think the rename is needed given this is an internal static function.

Reply:
> We now need to call `BIO_free(rctx->req)` in `OSSL_HTTP_REQ_CTX_free()`,
Heh, of course.

Reply:
The rename is not needed, but would be good to have because it makes clear that at this point the `BIO_free()` needs to be done only on error.